iceoryx_dust/vocabulary/include/iox/span.hpp

```diff
@@ -0,0 +1,451 @@
+// Copyright (c) 2022 - 2023 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_DUST_VOCABULARY_SPAN_HPP
+#define IOX_DUST_VOCABULARY_SPAN_HPP
+
+#include "iceoryx_hoofs/cxx/requires.hpp"
+#include "iox/detail/span_iterator.hpp"
+#include "iox/detail/uninitialized_array_type_traits.hpp"
+#include "iox/type_traits.hpp"
+#include "iox/uninitialized_array.hpp"
+
+#include <algorithm>
+#include <cassert>
+#include <cstdint>
+#include <cstdio>
+#include <limits>
+
+using namespace std;
+
+namespace iox
+{
+// constants
+constexpr uint64_t DYNAMIC_EXTENT = std::numeric_limits<uint64_t>::max();
+
+// Implementation of C++17 std::size() and std::data()
+
+/// @brief Returns c.size(), converted to the return type if necessary.
+/// @tparam C Type of the container
+/// @param c A container or view with a size member function
+/// @return The size of c
+template <typename Container>
+constexpr auto size(const Container& container) -> decltype(container.size());
+
+/// @brief Returns N
+/// @tparam T Type of the array
+/// @tparam N Size of the array
+/// @param An array
+/// @return Size of the array
+template <typename T, std::uint64_t N>
+constexpr std::uint64_t size(const T (&)[N]) noexcept;
+
+/// @brief Returns N
+/// @tparam T Type of the iox::UninitializedArray
+/// @tparam N Size of the iox::UninitializedArray
+/// @param An iox::UninitializedArray
+/// @return Size of the iox::UninitializedArray
+template <typename T, std::uint64_t N, template <typename, uint64_t> class Buffer>
+constexpr std::uint64_t size(const UninitializedArray<T, N, Buffer>&) noexcept;
+
+/// @brief Returns a pointer to the block of memory containing the elements of the range.
+/// @tparam C Type of the container
+/// @param c A container or view with a data() member function
```

Comment:
```suggestion
/// @tparam Container Type of the container
/// @param container A container or view with a data() member function
```

Please also change the other occurrences